src/DynamicData/Cache/SourceCache.cs

```diff
@@ -32,6 +32,7 @@ public SourceCache(Func<TObject, TKey> keySelector)
             }
 
             _innerCache = new ObservableCache<TObject, TKey>(keySelector);
+            KeySelector = keySelector;
```

Comment:
I would probably do a combined null check with assignment here

`KeySelector = keySelector ?? new ArgumentNullException(nameof(keySelector));`

This would require setting it above the `_innerCache`.